api/services/SyncService.js

```diff
@@ -41,6 +43,23 @@ function addVideo(video) {
   });
 }
 
+function addPlaylist(videos) {
+  let filteredVideos = videos.filter(video => {
```

Comment:
This is a poorly named variable. Some of the videos in `videos` are null because the video no longer exists, but is still in the playlist. I will rename this variable so it is clearer.